src/containers/Settings/UserSettings.js

```diff
@@ -3,34 +3,42 @@ import styles from './Settings.module.scss';
 import { compose } from 'redux';
 import { connect } from 'react-redux';
 import { FormSection } from 'explorer-components';
-import { rootUploadSelector } from 'public-modules/FileUpload/selectors';
-import { actions as uploadActions } from 'public-modules/FileUpload';
 import { actions as skillActions } from 'public-modules/Skills';
 import { actions as settingsActions } from 'public-modules/Settings';
+import { actions as settingsUIActions } from './reducer';
+import { settingsUISelector } from './selectors';
 import { skillsSelector } from 'public-modules/Skills/selectors';
 import { languagesSelector } from 'public-modules/Languages/selectors';
-import { settingsSelector } from 'public-modules/Settings/selectors';
+import {
+  profileImageUploadStateSelector,
+  settingsSelector
+} from 'public-modules/Settings/selectors';
 import { getCurrentUserSelector } from 'public-modules/Authentication/selectors';
 import { Field, reduxForm } from 'redux-form';
 import validators from 'utils/validators';
-import { ipfsToHttp } from 'utils/helpers';
 import { Cropper, Button, Text } from 'components';
-import { FormTextInput, FormSearchSelect, FormCheckbox } from 'form-components';
-import { UPLOAD_KEY } from './constants';
+import { FormCheckbox, FormTextInput, FormSearchSelect } from 'form-components';
 
 class UserSettingsComponent extends React.Component {
   constructor(props) {
     super(props);
 
-    const { ipfsHash, fileName } = props;
-    this.state = {
-      profileImage: ipfsHash ? ipfsToHttp(ipfsHash, fileName) : null
-    };
+    const { initialValues, setProfileImageUrls } = props;
+
+    const { smallProfileImageUrl, largeProfileImageUrl } = initialValues;
+
+    setProfileImageUrls(smallProfileImageUrl, largeProfileImageUrl);
   }
+
+  componentWillUnmount() {
+    this.props.resetProfileImageUrls();
```

Comment:
there was a bug (it here prior to this PR) where after uploading an image then navigating to a different page and returning the image would still be there.

Reply:
awesome catch.